### PR TITLE
Make the allinea easyblock search for the templates in the easyconfig paths

### PR DESCRIPTION
**Pull request by akesandgren**
Also add support for specifying a system.config to install.

This makes it possible to keep the templates in the sites easyconfig repos and also handles installing a system.config file if requested.

### CODE REVIEW

File: easybuild/easyblocks/a/allinea.py [hunk 1]
```diff
@@ -81,11 +82,34 @@ def install_step(self):
...
+        # copy system.config if requested
+	sysconf_path = os.path.join(self.installdir, 'system.config')
+	if self.cfg['sysconfig'] is not None:
+	    path = self.obtain_file(self.cfg['sysconfig'], extension=False)
+	    if path:
+		self.log.debug('system.config file %s found' % path)
+	    else:
+		raise EasyBuildError('No system.config file named %s found', sysconfig)
+            try:
+                shutil.copy(path, sysconf_path)
```
Comment (boegel) [inline easybuild/easyblocks/a/allinea.py]: @akesandgren please use `copy_file` (from `filetools` module) instead

File: easybuild/easyblocks/a/allinea.py [hunk 2]
```diff
@@ -81,11 +82,34 @@ def install_step(self):
...
+            except OSError, err:
+                raise EasyBuildError("Failed to copy %s to %s: %s", path, sysconf_path, err)
+	    try:
+		os.chmod(sysconf_path, 0444)
```
Comment (boegel) [inline easybuild/easyblocks/a/allinea.py]: @akesandgren please use `adjust_permissions` function for this?

File: easybuild/easyblocks/a/allinea.py [hunk 3]
```diff
@@ -81,11 +82,34 @@ def install_step(self):
...
         for templ in self.cfg['templates']:
+	    path = self.obtain_file(templ, extension='qtf')
+	    if path:
+		self.log.debug('Template file %s found' % path)
+	    else:
+		raise EasyBuildError('No template file named %s found', templ)
+
             try:
-                shutil.copy2(templ, templ_path)
+                shutil.copy(path, templ_path)
```
Comment (boegel) [inline easybuild/easyblocks/a/allinea.py]: @akesandgren why this change?

Comment (akesandgren) [inline easybuild/easyblocks/a/allinea.py]: If you mean the change from copy2 to copy, the installed file should have the same permissions as the rest of the installation, not whatever the template file had before the copy.

**Comment by akesandgren** [comment]
Anything else?

File: easybuild/easyblocks/a/allinea.py (hunk 3 above)
Comment (boegel) [inline easybuild/easyblocks/a/allinea.py]: ok, thanks for the clarification

File: easybuild/easyblocks/a/allinea.py [hunk 4]
```diff
@@ -81,11 +84,30 @@ def install_step(self):
...
             except OSError, err:
                 raise EasyBuildError("Failed to copy template %s to %s: %s", templ, templ_path, err)
 
+        # copy system.config if requested
+	sysconf_path = os.path.join(self.installdir, 'system.config')
```
Comment (boegel) [inline easybuild/easyblocks/a/allinea.py]: 🚨 tabs detected! 🚨

fixed in https://github.com/akesandgren/easybuild-easyblocks/pull/2

**Comment by boegel** [comment]
tested with existing Allinea easyconfigs, thanks @akesandgren!
